metadata-ingestion/src/datahub/ingestion/source/qlik_sense/qlik_api.py

```diff
@@ -74,6 +75,21 @@ def _get_dataset(self, dataset_id: str) -> Optional[QlikDataset]:
             )
         return None
 
+    def get_user_name(self, user_id: str) -> Optional[str]:
+        try:
+            if user_id in self.users:
```

Comment:
not necessary, but we could use `functools.lru_cache` here instead

Reply:
Use of functools.lru_cache creating some issue in api request.